dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ErrorDecoder.java

```diff
@@ -40,36 +48,66 @@
 enum ErrorDecoder {
     INSTANCE;
 
+    private static final Logger log = LoggerFactory.getLogger(ErrorDecoder.class);
     private static final ObjectMapper MAPPER = ObjectMappers.newClientObjectMapper();
 
     boolean isError(Response response) {
         return 300 <= response.code() && response.code() <= 599;
     }
 
-    RemoteException decode(Response response) {
+    RuntimeException decode(Response response) {
         // TODO(rfink): What about HTTP/101 switching protocols?
         // TODO(rfink): What about HEAD requests?
 
+        int code = response.code();
+        switch (code) {
+            case 308:
+                Optional<String> location = response.getFirstHeader(HttpHeaders.LOCATION);
+                if (location.isPresent()) {
+                    String locationHeader = location.get();
+                    try {
+                        return QosException.retryOther(new URL(locationHeader));
+                    } catch (MalformedURLException e) {
+                        log.error(
```

Comment:
Its a little subtle that we fall through if we fail to parse the location header since the 503 could have a serializable exception body and encoding properly set

Reply:
I don't think our utilities allow us to send 503 responses with serializable exception bodies, unless I'm missing something.

Agree that the fall-through is subtle, but I matched CJR behavior for now.

Reply:
ya that's fair